fe/fe-core/src/main/java/org/apache/doris/qe/ConnectProcessor.java

```diff
@@ -164,6 +164,13 @@ private void auditAfterExec(String origStmt, StatementBase parsedStmt, Data.PQue
                 .setQueryId(ctx.queryId() == null ? "NaN" : DebugUtil.printId(ctx.queryId()))
                 .setTraceId(spanContext.isValid() ? spanContext.getTraceId() : "");
 
+        // audit error msg
+        if (ctx.getState().getStateType() == MysqlStateType.ERR) {
+            ctx.getAuditEventBuilder().setErrType(ctx.getState().getErrType().name())
+                .setErrCode(ctx.getState().getErrorCode().name())
```

Comment:
In the following two cases, ErrType, ErrCode and ErrMsg will be empty or other values? 
1. if we got MysqlStateType.ERR, but getState().getErrorCode() not set;
2. the error code returned by getState().getErrorCode() not registered;

Reply:
Yes, this case may happed and cause null point exception in `ctx.getState().getErrorXXX().name()`.
Do you suggest adding a empty/null value judgment ?

Reply:
Yes I think it is necessary.

Reply:
Okeydokey